### PR TITLE
fix(block production): apply block before gossiping

### PR DESCRIPTION
**Pull request by danwt**
# PR Standards

## Opening a pull request should be able to meet the following requirements

---

Close https://github.com/dymensionxyz/dymint/issues/654

<-- Briefly describe the content of this pull request -->

For Author:

- [ ]  Targeted PR against correct branch
- [ ]  included the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
- [ ]  Linked to Github issue with discussion and accepted design
- [ ]  Targets only one github issue
- [ ]  Wrote unit and integration tests
- [ ]  All CI checks have passed
- [ ]  Added relevant `godoc` [comments](https://blog.golang.org/godoc-documenting-go-code)

---

For Reviewer:

- [ ]  confirmed the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
- [ ]  Reviewers assigned
- [ ]  confirmed all author checklist items have been addressed

---

After reviewer approval:

- [ ]  In case targets main branch, PR should be squashed and merged.
- [ ]  In case PR targets a release branch, PR should be rebased.

### CODE REVIEW

File: block/produce.go [hunk 1]
```diff
@@ -144,12 +144,12 @@ func (m *Manager) produceBlock(ctx context.Context, allowEmpty bool) error {
...
 	}
 
-	// Gossip the block as soon as it is produced
-	if err := m.gossipBlock(ctx, *block, *commit); err != nil {
+	if err := m.applyBlock(ctx, block, commit, blockMetaData{source: producedBlock}); err != nil {
 		return err
 	}
 
-	if err := m.applyBlock(ctx, block, commit, blockMetaData{source: producedBlock}); err != nil {
+	// Gossip the block as soon as it is produced
```
Comment (mtsitrin) [inline block/produce.go:151]: remove wrong comment

Comment (mtsitrin) [inline block/produce.go:151]:
```suggestion
	// Gossip the block after it's been commited
```

**Review by zale144 — APPROVED** [review]

**Comment by omritoptix** [comment]
@danwt just to make sure we have an issue on getting this out of the block production function all together as an error in the gossiping the block shouldn't be related to the block production.

**Comment by danwt** [comment]
@omritoptix yep sure, one thing at a time though!

**Review by omritoptix — APPROVED** [review]

**Pull request merged**